charts/prometheus-blackbox-exporter/templates/deployment.yaml

```diff
@@ -87,7 +90,7 @@ spec:
           resources:
 {{ toYaml .Values.resources | indent 12 }}
           ports:
-            - containerPort: {{ .Values.service.port }}
+            - containerPort: {{ .Values.port }}
```

Comment:
Why the change?
Now port is configurable twice?

Reply:
If you add a oAuthProxy in front, you need to differentiate the service port and the container port, so there is not direct access to the blackbox.
In Prometheus-pushgateway, the port if hard coded [1], maybe we can hard code it too ?

[1] https://github.com/prometheus-community/helm-charts/blob/a72459c5a1a5672c0bcd812a5a078ec4517693d2/charts/prometheus-pushgateway/templates/deployment.yaml#L40

Reply:
Why don't you add a targetPort key to the service and make it configurable by a variable like `targetPort: {{ .Values.service.targetPort }}`?
With this you could set the proxy to be accessed by the service instead of the alertmanager container.

Reply:
OK, I did as you suggested. I hope that works for you now.